src/lib/UI/Module/ContentTree/NodeFactory.php

```diff
@@ -35,27 +35,33 @@ final class NodeFactory
     private $maxDepth;
 
     /** @var string[] */
-    private $contentTypeIdentifierIgnoreList;
+    private $allowedContentTypes;
+
+    /** @var string[] */
+    private $ignoredContentTypes;
 
     /**
      * @param \eZ\Publish\API\Repository\SearchService $searchService
      * @param int $displayLimit
      * @param int $childrenLoadMaxLimit
      * @param int $maxDepth
-     * @param array $contentTypeIdentifierIgnoreList
+     * @param array $allowedContentTypes
+     * @param array $ignoredContentTypes
      */
     public function __construct(
         SearchService $searchService,
         int $displayLimit = 20,
         int $childrenLoadMaxLimit = 100,
         int $maxDepth = 10,
-        array $contentTypeIdentifierIgnoreList = []
+        array $allowedContentTypes = [],
+        array $ignoredContentTypes = []
     ) {
         $this->searchService = $searchService;
         $this->displayLimit = $displayLimit;
         $this->childrenLoadMaxLimit = $childrenLoadMaxLimit;
         $this->maxDepth = $maxDepth;
-        $this->contentTypeIdentifierIgnoreList = $contentTypeIdentifierIgnoreList;
+        $this->allowedContentTypes = $allowedContentTypes;
+        $this->ignoredContentTypes = $ignoredContentTypes;
```

Comment:
The number of configuration values which are passed via constructor is growing dangerously so might be good idea to introduce dedicated class representing configuration. 

```php
final class NodeFactorySettings {

    /** @var int **/    
    private $displayLimit;
    /** @var int **/
    private $childrenLoadMaxLimit;
    /** @var int **/
    private $maxDepth;
    /** @var string[] */
    private $allowedContentTypes;
    /** @var string[] */
    private $ignoredContentTypes;

    // ... 
}
```

Applying this approach will also simplify implementation of `EzSystems\EzPlatformAdminUi\UI\Config\Provider\Module\ContentTree`

Reply:
Do you want to create a service from `NodeFactorySettings`? I don't agree with this approach - a service from basically a ValueObject does not convince me. I'd rather pass `ConfigResolver` here and use it to get settings.